### PR TITLE
General and search utils tests

### PR DESCRIPTION
**Pull request by FalseR20**

### CODE REVIEW

File: tests/test_utils/test_generation_utils.py [hunk 1]
```diff
@@ -3,3 +3,117 @@
...
+    def test_create_structure(self):
+        element1 = create_node(sc_types.NODE_CONST)
+        element2 = create_node(sc_types.NODE_CONST)
+        set_node = create_structure(element1, element2)
+        search_results = client.template_search(_create_set_template(set_node, element1, element2))
+        assert search_results
+        assert search_results[0].addrs[0].value == set_node.value
+
+
+def _create_oset_template(set_node: ScAddr, start_element: ScAddr, other_element: ScAddr) -> ScTemplate:
```
Comment (AlexandrZagorskiy) [inline tests/test_utils/test_generation_utils.py:64]: It's not clear, this func create only template as a code, or function generate construction in memory. What you think about:
```suggestion
def _oriented_set_template(set_node: ScAddr, start_element: ScAddr, other_element: ScAddr) -> ScTemplate:
```

Comment (AlexandrZagorskiy) [reply tests/test_utils/test_generation_utils.py:64]: _oset_ name not correlate with _oriented_set_ for me, so maybe you can use more clarifying naming

File: tests/test_utils/test_search_utils.py [hunk 2]
```diff
@@ -3,3 +3,56 @@
...
+        set_node = create_node(sc_types.NODE_CONST)
+        elements = [
+            create_node(sc_types.NODE_CONST),
+            create_node(sc_types.NODE_CONST),
+            create_node(sc_types.NODE_CONST),
+        ]
+        wrap_in_set(set_node, *elements)
+        search_results = get_set_elements(set_node)
+        elements.reverse()
+        assert are_elements_equal(search_results, elements)
```
Comment (AlexandrZagorskiy) [inline tests/test_utils/test_search_utils.py:28]: Specify test, I want to be sure that construction forms correctly with concrete sequence

**Review by AlexandrZagorskiy — CHANGES_REQUESTED** [review]

File: tests/test_utils/test_generation_utils.py (hunk 1 above)
Comment (FalseR20) [reply tests/test_utils/test_generation_utils.py:64]: I'll change to `_create_oriented_set_template`

File: src/sc_kpm/utils/search_utils.py [hunk 3]
```diff
@@ -16,37 +16,34 @@
...
-        else:
-            keynodes = ScKeynodes()
-            elem_search_result = search_role_relation_template(set_node, keynodes[CommonIdentifiers.RREL_ONE.value])
-        if elem_search_result is None:
-            return None
-        elements.append(elem_search_result.get(ScAlias.ELEMENT.value))
-        return elem_search_result.get(ScAlias.ACCESS_EDGE.value)
-
     elements = []
-    elements_count = get_set_power(set_node)
```
Comment (AlexandrZagorskiy) [inline src/sc_kpm/utils/search_utils.py:42]: Are you sure that we can remove calling of this method? Write tests for empty sets to ensure about it

**Review by AlexandrZagorskiy — CHANGES_REQUESTED** [review]

File: src/sc_kpm/utils/search_utils.py (hunk 3 above)
Comment (FalseR20) [reply src/sc_kpm/utils/search_utils.py:42]: Yeah. We needn't get power of set if we seach only one rrel edge and if it don't exist function returns

**Review by AlexandrZagorskiy — APPROVED** [review]

**Pull request merged**